packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -497,13 +497,16 @@ export class ProfilesCache {
     /**
      * Used for extenders to register with Zowe Explorer that do not need their
      * profile type in the existing MVS, USS, and JES
-     * @param {string} profileTypeName Type of Profile, defaults to "zosmf" if nothing passed.
+     * @param {string} profileTypeName Type of Profile
      *
      * @returns {void}
      */
     public async asyncRegisterCustomProfilesType(profileTypeName: string): Promise<void> {
         const existsV1Profile = fs.existsSync(path.posix.join(`${os.homedir()}/.zowe/profiles/${profileTypeName}`));
-        const existsV2Profile = (await this.getProfileInfo()).usingTeamConfig;
+        const profileInfo = await this.getProfileInfo();
+        const existsV2Profile =
+            profileInfo?.usingTeamConfig &&
+            profileInfo.getAllProfiles().find((profile) => profile.profType === profileTypeName);
```

Comment:
@KutluOzel-b this will only work if a profile of that type already exists, it there a call we can make to imperative to check against the type's existence in the configs schema?

Reply:
@JillieBeanSim Isn't that we check for in the first place , that if the specified profile type is already exists or not ? I'm having lack of knowledge here. We can get loaded profile from cli too though but I'm can't tell if this is the right call.

Reply:
I am starting to think that we shouldn't require the check for v1 or v2 profiles types and just have this method use the final line `this.allExternalTypes.add(profileTypeName);` to register the type and leave any checks for existing profiles be done by extenders using this.  What do you think @KutluOzel-b

Reply:
@JillieBeanSim according to the need of extenders since profileTypeName is optional or it is irrelevant to have one more right to say, I agree with you that we don't need to check any existing profiles but to add  directly via `this.allExternalTypes.add(profileTypeName);`